priv/perf/apps/load_test/lib/connection/connection_defaults.ex

```diff
@@ -34,11 +34,13 @@ defmodule LoadTest.Connection.ConnectionDefaults do
   """
   def pool_name(), do: :perf_pool
 
+  # Don't automatically retry on error
+  # It _can_ sometimes be useful to retry though, so if you need it return true here
   defp retry?() do
     fn
-      {:ok, %{status: status}} when status in 500..599 -> true
+      {:ok, %{status: status}} when status in 500..599 -> false
```

Comment:
I think instead of doing this -> false thing, better ways are:
1. restrict this to 502 if you agree we want to ignore infra failure from our load test and we really don't want our test to stop at the middle (especially if it takes some time to run). I am guessing you're avoiding retry because the previous issue that the child chain does accepts the tx but timed out and retry would just not make sense. That would be 500 if I get it right.
1. Or if you don't agree with above, then set this with config/env_var, and describe how we do this in readme instead of the comment here. We default to `false` if you have experienced that most of the time we don't want to retry.

Reply:
My thinking is that by default we don't want to ignore _any_ failures, because finding failures is the main reason for running the tests.

I get that we might want to keep the tests running regardless of errors, but if there is a known infra problem (such as the 502 error) and you want to ignore it and run the tests anyway, then I think you need to explicitly do that and understand how the error will affect the test results.

I'm not sure how to set that via config though, because you may want to retry on a status code (or a range of status codes)
```
{:ok, %{status: status}} when status = 502 -> true
```
or a particular error e.g.
```
{:error, "reason: timeout"} -> true
```

Do you have any idea?
Or should I just put a better explanation with some examples into the readme?

Reply:
>  because finding failures is the main reason for running the tests.

Out of scope of this PR I would say, but better way I can think of is collect this failure metric and let the test to continue running. Usually the test should be run for a certain period, so the cost of re-run is quite high....for an engineer to wait 😛 We just want to know whether it has failed or not but we are not fixing with the load test anyway so we can continue to test the performance. Also, I think it might be interesting to monitor how our system works after some error occurs, does the error rate explodes after the first one or it auto recovers well.

> Do you have any idea?

lol, okay you're definitely thinking sth harder then I expect haha. I was just suggesting to either allow retry on 5xx or not with env_var or config. I guess we can start by either simply adding flag to turn retry on or let the env var passing in the list of status code we retry? (so turning off we pass in empty list)

Reply:
Failure rate! Yeah, that would be really cool!